rero_ils/modules/documents/dojson/contrib/jsontodc/model.py

```diff
@@ -0,0 +1,268 @@
+        self['dates'] = dates
+    if publishers:
+        # write the publishers directly into self
+        self['publishers'] = publishers
+
+
+@dublincore.over('types', '^type')
+@utils.for_each_value
+@utils.ignore_value
+def json_to_types(self, key, value):
+    """Get types data."""
+    main_type = value.get('main_type')
+    subtype_type = value.get('subtype')
+    if subtype_type:
+        return ' / '.join([_(main_type), _(subtype_type)])
+    else:
+        return _(main_type)
+
+
+@dublincore.over('identifiers', '^identifiedBy')
+@utils.for_each_value
+@utils.ignore_value
+def json_to_identifiers(self, key, value):
+    """Get identifiers data."""
+    itype = value.get('type')
+    identifier_value = value.get('value')
+    source = value.get('source')
+    if source:
+        return f'{itype}|{identifier_value}({source})'
+    return f'{itype}|{identifier_value}'
+
+
+@dublincore.over('relations',
+                 '^(issuedWith|otherEdition|otherPhysicalFormat|precededBy|'
+                 'relatedTo|succeededBy|supplement|supplementTo)')
+@utils.for_each_value
+@utils.ignore_value
+def json_to_relations(self, key, value):
+    """Get relations data."""
+    label = value.get('label')
+    if label:
+        return label
+    else:
+        # TODO: make shure the $ref was replaced and had a _text field.
+        return value.get('_text')
+
+
+@dublincore.over('subjects', '^subjects')
+@utils.for_each_value
+@utils.ignore_value
+def json_to_subject(self, key, value):
+    """Get subject data."""
+    result = ''
+    subject_type = value.get('type')
+    if subject_type in ['bf:Person', 'bf:Organisation', 'bf:Place']:
+        # TODO: set the language
+        authorized_access_point = get_contribution_localized_value(
+            contribution=value,
+            key='authorized_access_point',
+            language=dublincore.language
+        )
+        if authorized_access_point:
+            result = authorized_access_point
+        else:
+            result = value.get('preferred_name')
+    elif subject_type == 'bf:Work':
+        work = []
+        creator = value.get('creator')
+        if creator:
+            work.append(creator)
+        work.append(value.get('title'))
+        result = '. - '.join(work)
+    elif subject_type in ['bf:Topic', 'bf:Temporal']:
+        result = value.get('term')
+    return result or None
+
+
+# @dublincore.over('coverage', '^coverage')
+# @utils.ignore_value
+# def json_to_coverage(self, key, value):
+#     """Get coverage data."""
+
+
+# @dublincore.over('format', '^formats')
+# @utils.ignore_value
+# def json_to_formats(self, key, value):
+#     """Get dates data."""
+
+
+# @dublincore.over('rights', '^rights')
+# @utils.ignore_value
+# def json_to_rights(self, key, value):
+#     """Get rights data."""
+
+
+# @dublincore.over('sources', '^sources')
+# @utils.ignore_value
+# def json_to_sources(self, key, value):
+#     """Get sources data."""
```

Comment:
Should this be removed?

Reply:
Is in specification and should be implemented later.